### PR TITLE
Added cypress tests for modal on close promise

### LINKED ISSUE #3675: Add test for modal on close promise
Add a test to test the functionality of openAsModal on close promise functionality added [here](https://github.com/SAP/luigi/pull/3672)

Could be a Cypress e2e test

### PR DESCRIPTION
**Pull request by VincentUllal**
**Description**

**Related issue(s)**

Fixes #3675

### CODE REVIEW

File: test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    it('print stuff out so, I can check what functions are available etc.', () => {
+      cy.window().then(win => {
+        cy.log(win)
+        cy.log("----win.Luigi----")
+        cy.log(win.Luigi)
+        cy.log("----win.Luigi.navigation()----")
+        cy.log(win.Luigi.navigation())
+        cy.log("----win.Luigi.navigation().openAsModal() without cy.wrap----")
+        cy.log(win.Luigi.navigation().openAsModal('/projects/pr2'))
```
Comment (ndricimrr) [inline test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js:17]: I think this task for now should focus on LuigiClient rather than Luigi Core.
win.Luigi seems like an attempt at accessing Luigi Core API.
There are two mirrored functions on both Core and Client:
1 - [openAsModal on Core](https://docs.luigi-project.io/docs/luigi-core-api/?section=examples-28)
2 - [openAsModal on Client](https://docs.luigi-project.io/docs/luigi-client-api/?section=examples-30)

Here you focus on 2 instead.

File: test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        const p = win.Luigi.navigation().openAsModal('/projects/pr2')
```
Comment (ndricimrr) [inline test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js:19]: For future reference `win.Luigi.navigation()....` is something we have done in the past but its bad practice to access Luigi Object or any other JS Object on a Cypress test.

Instead you would add an HTML Element =(button or whatever) and attach an onclick method on the code (Our Angular E2E App or other) that runs the Luigi API call.
The Luigi API call itself usually affects the DOM: Ex: Opens a modal that is present in the DOM.

So in short Cypress is good for detecting Visual DOM changes, so keep that in mind when writing tests with it.

File: test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        cy.wrap(p).then((res, rej) => {   // Question: if .then() is only executed after the promise is fulfilled, why is this part executed even though the modal has not been closed yet?
```
Comment (ndricimrr) [inline test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js:20]: I think this would not work because
`Luigi.navigation().openAsModal(...).then`  does not exist.

I think the [openAsModal](https://docs.luigi-project.io/docs/luigi-core-api/?section=examples-28) on Luigi Core rather takes an extra parameter as a onCloseCallback function instead of returning a promise

**Review by ndricimrr — COMMENTED** [review]
I think you are aiming for the wrong function here.

Focus on 
```
LuigiClient.linkManager().openAsModal('projects/pr1/users', {title:'Users', size:'m'}).then((res) => {
// console.log
}
```

File: test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js (hunk 2 above)
Comment (VincentUllal) [reply test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js:19]: Ah I see, good to know :)

File: test/e2e-test-application/src/app/project/project.component.ts [hunk 4]
```diff
@@ -128,6 +128,25 @@ export class ProjectComponent implements OnInit, OnDestroy {
...
+  openAsModal() {
+    linkManager().openAsModal('/projects/pr2/settings', {
+      title: 'microfrontend in a modal',
+      size: 'm',
+    }).then(() => {
+      const newListEl = document.createElement('li');
+      newListEl.textContent = 'promise resolved!';
+      newListEl.id = "promiseResolved";
+      newListEl.className = 'fd-list__item';
+      document.getElementById('promiseTest').appendChild(newListEl);
```
Comment (ndricimrr) [inline test/e2e-test-application/src/app/project/project.component.ts:140]: this is a little overcomplicated.
I think just one line here: `document.getElementById('promiseTest').innerHTML='promise resolved!' `
would be enough.

File: test/e2e-test-application/src/app/project/project.component.ts [hunk 5]
```diff
@@ -128,6 +128,25 @@ export class ProjectComponent implements OnInit, OnDestroy {
...
+    }).catch(() => {
+      const newListEl = document.createElement('li');
+      newListEl.textContent = 'promise rejected';
+      newListEl.id = "promiseRejected";
+      newListEl.className = 'fd-list__item';
+      document.getElementById('promiseTest').appendChild(newListEl);
+    });
+  }
```
Comment (ndricimrr) [inline test/e2e-test-application/src/app/project/project.component.ts:148]: I can't think of any ways how/or when this would reject. I would omit the catch clause altogether. the then clause should be enough 👍

Comment (VincentUllal) [reply test/e2e-test-application/src/app/project/project.component.ts:148]: I added this just in case there is a reject. I'll remove it👍

**Review by ndricimrr — APPROVED** [review]
Looks good 👍
